src/main/cpp/win_fsnotifier.cpp

```diff
@@ -51,99 +176,96 @@ void handlePathChanged(watch_details_t *details, FILE_NOTIFY_INFORMATION *info)
     } else if (info->Action == FILE_ACTION_MODIFIED) {
         type = FILE_EVENT_MODIFIED;
     } else {
-        wprintf(L"~~~~ Unknown event 0x%x for %ls\n", info->Action, changedPath);
+        wprintf(L"~~~~ Unknown event 0x%x for %ls\n", info->Action, changedPath.c_str());
         type = FILE_EVENT_UNKNOWN;
     }
 
-    bool watching = false;
-    for (int i = 0; i < details->watchedPathCount; i++) {
```

Comment:
suggestion: `for (auto&& watchedPath : details->watchedPaths) {}` or sg. similar

Reply:
or even better, `std::find_if()` and an in-place lambda checker